client/app/api/course/Disbursement.ts

```diff
@@ -0,0 +1,102 @@
+import { AxiosResponse } from 'axios';
+import { CourseUserListData } from 'types/course/courseUsers';
+import {
+  CourseGroupListData,
+  ForumDisbursementFilters,
+  ForumDisbursementFilterParams,
+  ForumDisbursementUserData,
+} from 'types/course/disbursement';
+import BaseCourseAPI from './Base';
+
+export default class DisbursementAPI extends BaseCourseAPI {
+  _getUrlPrefix(): string {
+    return `/courses/${this.getCourseId()}/users/disburse_experience_points`;
+  }
+
+  _getForumUrlPrefix(): string {
+    return `/courses/${this.getCourseId()}/users/forum_disbursement`;
+  }
+
+  /**
+   * Fetches disbursement data.
+   */
+  index(): Promise<
+    AxiosResponse<{
+      currentGroup: CourseGroupListData | null;
+      courseGroups: CourseGroupListData[];
+      courseUsers: CourseUserListData[];
+    }>
+  > {
+    return this.getClient().get(this._getUrlPrefix());
+  }
+
+  /**
+   * Fetches disbursement data.
+   */
+  forumIndex(): Promise<
+    AxiosResponse<{
+      filters: ForumDisbursementFilters;
+      forumUsers: ForumDisbursementUserData[];
+    }>
+  > {
+    return this.getClient().get(this._getForumUrlPrefix());
+  }
+
+  /*
+   * Fetches disbursement data with group filter.
+   */
+  fetchForumIndexWithFilter(params: ForumDisbursementFilterParams): Promise<
```

Comment:
fetchForumDisbursementIndexWithFilter

Reply:
I think this can be combined with forumDisbursementIndex and make params optional.